docs/DeveloperManuals/PluginImplementation.md

```diff
@@ -17,8 +17,15 @@ A plugin may extend DevLake's capability in three ways:
 2. Transforming/enriching existing data
 3. Exporting DevLake data to other data systems
 
+## Types of plugins
 
-## How do plugins work?
+There are, as of now, support for two types of plugins:
+
+1. __*Conventional plugins*__: These are the primary type of plugins used by Devlake, and require the developer to write the most amount of code starting from fetching (collecting) data from data sources to converting them into our normalized datamodels and storing them.
+2. __*Singer-spec plugins*__: These plugins utilize [Singer-taps](https://www.singer.io/) to retrieve data from data-sources thereby eliminating the developer's burden of writing the collection logic. More on them [here](#how-do-singer-spec-plugins-work).
+
+
+## How do conventional plugins work?
```

Comment:
I chose "conventional" as way of referring to the original plugins. Open to suggestions of better adjectives